src/test/java/org/ays/auth/tests/RoleCreateTest.java

```diff
@@ -0,0 +1,153 @@
+    public void createRoleWithInvalidRoleName(String name, AysErrorMessage errorMessage, String field, String type) {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setName(name);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .spec(AysResponseSpecs.subErrorsSpec(errorMessage, field, type));
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleWithNullRoleName() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setName(null);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .body("subErrors[0].message", equalTo("must not be blank"));
+    }
+
+    @Test(groups = {"Regression"}, dataProvider = "invalidPermissionIds", dataProviderClass = AysDataProvider.class)
+    public void createRoleWithInvalidPermissionIdList(List<String> permissionIds, AysErrorMessage errorMessage,
+                                                      String field, String type) {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(permissionIds);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .spec(AysResponseSpecs.subErrorsSpec(errorMessage, field, type));
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleWithNullPermissionId() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(null);
+
+        Response response = RoleEndpoints.create(role, accessToken);
+        response.then()
+                .spec(AysResponseSpecs.expectBadRequestResponseSpec())
+                .body("subErrors[0].message", equalTo("must not be empty"));
+    }
+
+    @Test(groups = {"Regression"})
+    public void createRoleWithDuplicatePermissionIds() {
+
+        RoleCreatePayload role = RoleCreatePayload.generate();
+        role.setPermissionIds(List.of("17dd50f6-61fe-4a30-a136-d9b80649e7fe", "17dd50f6-61fe-4a30-a136-d9b80649e7fe"));
+        RoleEndpoints.create(role, accessToken);
+        String roleId = RoleDataSource.findLastCreatedRoleIdByInstitutionId(AysConfigurationProperty
+                .TestVolunteerFoundation.ID);
+
+        List<String> permissionIds = PermissionDataSource.getPermissionIdsFromLastCreatedRole(roleId);
+        List<String> expectedPermissionIds = List.of("17dd50f6-61fe-4a30-a136-d9b80649e7fe");
+
+        Assert.assertEquals(permissionIds.size(), 1, "Permission IDs should be unique in the database.");
+        Assert.assertEquals(permissionIds, expectedPermissionIds);
+    }
```

Comment:
Bu test ile aynı permisson id ile rol oluşturma hedeflenmişse eğer bunun api danda kontrolü yapılmalı. Denediğimde api da iki aynı permission id ile istek atabiliyoruz ama veri tabanına tek olarak kaydediliyor. Bence api'dan böyle bir istek denendiğinde success alınmamalı. Bu konuyu bi be-support'a sormakta fayda görüyorum. 
Ek olarak permission id leri böyle el ile yazmasak daha iyi olabilir.

Reply:
@cagla88 katiliyorum ama bununla ilgili manual testlerde boyle olması gerektiğiyle ilgili madde var checklistlerde, "Yapılan istekte "permissionIds" listesi ayni permission ID tekrarlarından oluşuyorsa hata vermeden  kayıt yapar. DB'de role karsılık tek permission gözlenir." seklinde. Dolayisiyla boyle karar verilmiş demek ki diye düşünmüştüm, @agitrubard bu durum değişecek mi acaba ?

Reply:
@htczrrsz bu durumu bilmiyordum ama böyle söylenmişse sorun yok.

Reply:
@cagla88 detayli konuşuldu mu bilmiyorum acikcasi ama checklist'te Nebahat boyle belirtmiş